### ISSUE TITLE
[Feature] X-Response-Time header

### ISSUE THREAD

**Issue by andreespirela** [mandarineorg/mandarinets#192]
As a user I want to have the option to enable/disable a request time middleware which would add the `X-Response-Time` header to the response.

**Comment by Suyashtnt** [comment]
here is the code needed for it to work:
```ts
@Middleware()
export class ResponseTime implements MiddlewareTarget {
	date = Date.now();
	public onPreRequest(): boolean {
		this.date = Date.now();
		return true;
	}

	public onPostRequest(@ResponseParam() response: Response): void {
		this.date = Date.now() - this.date;
		response.headers.set('X-Response-Time', this.date.toString());
		console.log(`done. It took ${this.date}ms`);
	}
}
```

**Comment by andreespirela** [comment]
@Suyashtnt While we don't use any mandarine-powered component in the core (we don't use any mandarine feature for the core), I will make use of this logic. Very much appreciated it.

**Comment by andreespirela** [comment]
Closed in #201 #202
